arangod/RestServer/LanguageCheckFeature.cpp

```diff
@@ -151,21 +152,13 @@ ErrorCode writeLanguage(arangodb::ArangodServer& server, std::string_view lang,
   return TRI_ERROR_NO_ERROR;
 }
 
-std::tuple<std::string, LanguageType> getOrSetPreviousLanguage(
-    arangodb::ArangodServer& server, std::string_view collatorLang,
-    LanguageType currLangType) {
+std::tuple<std::string, LanguageType> getPreviousLanguage(
+    arangodb::ArangodServer& server) {
   std::string prevLanguage;
   LanguageType prevType;
-  arangodb::Result res = ::readLanguage(server, prevLanguage, prevType);
+  [[maybe_unused]] auto res = ::readLanguage(server, prevLanguage, prevType);
```

Comment:
There are code paths in `::readLanguage` which do not initialize `prevType`, so I suggest to initialize `prevType` here before calling `::readLanguage`.

Reply:
We probably want to initialize it with type `EMPTY`.